Input: synaptics-rmi4- Add a new feature for Forepad.

myangsynapticscom opened the pull request:
Forcepad devices will use F21, for click simulation due to lack of a metal button, so we add F21 support to make forcepad support click function.

myangsynaptics committed:
Input: synaptics-rmi4- Add a new feature for Forepad.

Forcepad devices will use F21, for click simulation
due to lack of a metal button, so we add F21 support
to make forcepad support click function.

Signed-off-by: Marge Yang <Marge.Yang@tw.synaptics.com>

vhuag commented on drivers/input/rmi4/rmi_f21.c:
```diff
@@ -0,0 +1,125 @@
+			dev_warn(&fn->dev, "f21 interrupted, but data is missing\n");
+			return IRQ_HANDLED;
+		}
+		memcpy(f21->data_regs, drvdata->attn_data.data, RMI_f21_INPUT_REPORT_DATA_SIZE);
```
F21 attention data is not matched data register totally, so could we use other buffer for this check and don't use data_regs instead?

vhuag commented:
please also fix commit decription typo

myangsynaptics committed:
Input: synaptics-rmi4- Add a new feature for Forcepad.

Forcepad devices will use F21, for click simulation
due to lack of a metal button, so we add F21 support
to make forcepad support click function.

Signed-off-by: Marge Yang <Marge.Yang@tw.synaptics.com>

vhuag commented:
This version is better.  it looks good to me.
